### PR TITLE
[new] add jquery.smartmenus@1.0.0 with git auto-update, close #5095

### LINKED ISSUE #5095: [Request] jquery.smartmenus.js
Repository: https://github.com/vadikom/smartmenus

### PR DESCRIPTION
**Pull request by Amomo**
Hi @maruilian11, I add a new lib with git auto-update from the request #5095 
This lib contains some customized addons and css files.
Some of them are lack of minimized, so I attach on minified version.
I record them in the "requiredFile" field.
Would you mind checking this PR for me? Thank you~ :-)

### CODE REVIEW

**Comment by maruilian11** [comment]
i think it's ok. @PeterDaveHello

**Comment by PeterDaveHello** [comment]
Why don't you just use `**/*` in the filemap?

**Comment by maruilian11** [comment]
@PeterDaveHello but there are some <code>.scss</code> under dist, do we need them, too?

**Comment by PeterDaveHello** [comment]
Of course not.

**Comment by Amomo** [comment]
I use `css/**/*.css`, `css/*.css` to avoid those `.scss` files.

File: ajax/libs/jquery.smartmenus/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/vadikom/smartmenus.git",
+    "basePath": "dist/",
+    "files": [
+      "addons/**/*",
+      "css/**/*.css",
+      "css/*.css",
```
Comment (PeterDaveHello) [inline ajax/libs/jquery.smartmenus/package.json]: Duplicated with line 29

**Comment by Amomo** [comment]
Oh, it's my fault!
There is a .css file under css/, and I forget to contain it's minimized file in the commit.
That's why I think we should also use `"css/*.css"`.
I have done the correction.
@maruilian11 could you help me check it again? many thanks! :-D

**Comment by maruilian11** [comment]
@PeterDaveHello i think it's ok.

**Comment by PeterDaveHello** [comment]
Line 30 is duplicated with line 29 ...

**Comment by Amomo** [comment]
Oh! I got the point finally. Thank you, Peter.
@maruilian11 I remove the duplicated code in package.json.
Could you check this PR again? Many thanks to you! :-D

File: ajax/libs/jquery.smartmenus/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "type": "git",
+      "url": "https://github.com/vadikom/smartmenus"
+    }
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/vadikom/smartmenus.git",
+    "basePath": "dist/",
+    "files": [
+      "addons/**/*",
```
Comment (PeterDaveHello) [inline ajax/libs/jquery.smartmenus/package.json]: I'll use only `**/*.+(css|js)` to grab them all.

File: ajax/libs/jquery.smartmenus/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      "css/**/*.css",
+      "jquery.smartmenus*.js"
+    ]
```
Comment (PeterDaveHello) [inline ajax/libs/jquery.smartmenus/package.json]: I'll use only `**/*.+(css|js)` to grab them all.

Comment (Amomo) [inline ajax/libs/jquery.smartmenus/package.json]: Wow! Thank you for point out this.
This expression is more clear and intelligent!
I will take that. :-D

**Comment by maruilian11** [comment]
@Amomo i think it's ok. can u rebase it, then tag me ?
sorry, i didn't check it at the first time.

**Comment by Amomo** [comment]
Hi @maruilian11 , I have rebased it~

**Comment by maruilian11** [comment]
@PeterDaveHello i think it's ok.

**Pull request merged**